packages/flutter_tools/lib/src/android/android_sdk.dart

```diff
@@ -229,6 +232,10 @@ class AndroidSdkVersion implements Comparable<AndroidSdkVersion> {
   String get apksignerPath => getBuildToolsPath('apksigner');
 
   List<String> validateSdkWellFormed({bool requireApkSigner = true}) {
+    if (buildToolsVersion.major < minimumAndroidSdkVersion) {
```

Comment:
OK, sounds like we have some later cleanup to do. Logged https://github.com/flutter/flutter/issues/8525 for that.